apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/commons/AbstractSamhsaMatcherTest.java

```diff
@@ -0,0 +1,812 @@
+package gov.cms.bfd.server.war.commons;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anySet;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import gov.cms.bfd.model.codebook.data.CcwCodebookVariable;
+import gov.cms.bfd.server.war.adapters.CodeableConcept;
+import gov.cms.bfd.server.war.adapters.Coding;
+import gov.cms.bfd.server.war.adapters.DiagnosisComponent;
+import gov.cms.bfd.server.war.adapters.FhirResource;
+import gov.cms.bfd.server.war.adapters.ItemComponent;
+import gov.cms.bfd.server.war.adapters.ProcedureComponent;
+import gov.cms.bfd.server.war.utils.ReflectionTestUtils;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.mockito.ArgumentCaptor;
+
+@RunWith(Enclosed.class)
+public class AbstractSamhsaMatcherTest {
+
+  private static final IllegalStateException INVOCATION_EXCEPTION =
+      new IllegalStateException("Method was not called with expected arguments for test");
+
+  @FunctionalInterface
+  public interface SamhsaFilterMethod<T> {
+    boolean apply(AbstractSamhsaMatcher<IBaseResource> matcher, T coding);
+  }
+
+  public static class NonParameterizedTests {
+
+    @Test
+    public void shouldReturnColumnValues() {
+      List<String> expected = List.of("1", "2", "3", "4");
+      List<String> actual =
+          AbstractSamhsaMatcher.resourceCsvColumnToList("samhsa_codes_test_file.csv", "columnB");
+
+      assertEquals(expected, actual);
+    }
+
+    @Test
+    public void shouldReturnTrueForSamhsaProcedureComponent() {
+      // unchecked - This is ok for making a mock.
+      //noinspection unchecked
+      AbstractSamhsaMatcher<IBaseResource> matcherSpy = spy(AbstractSamhsaMatcher.class);
+
+      ProcedureComponent mockComponentA = mock(ProcedureComponent.class);
+      ProcedureComponent mockComponentB = mock(ProcedureComponent.class);
+
+      doThrow(INVOCATION_EXCEPTION)
+          .when(matcherSpy)
+          .isSamhsaIcdProcedure(any(ProcedureComponent.class));
```

Comment:
It's a safety net incase the test is set up incorrectly.  By default, any `spy` object will execute the default logic unless a specific mock circumstance is encountered, meaning it will mock the return instead.  This throw prevents the default logic from ever running, avoiding accidental passes on the test.  If we wanted to document each one, we could, just wondering if it would be over cluttering the tests.

Reply:
Gotcha. Maybe make a private method to set up the matcherSpy, include this there, and document why in that method doc? Then we can reduce duplicated code and still document why it needs to exist.

```
/** 
  * Sets up the spy used to simulate the SamhsaMatcher under test. To prevent improper test setup,
  * sets up an exception to be thrown when the default isSamhsaIcdProcedure functionality is called.
**/
private AbstractSamhsaMatcher<IBaseResource> setupTestSamhsaMatcher() {
    AbstractSamhsaMatcher<IBaseResource> matcherSpy = spy(AbstractSamhsaMatcher.class);
    doThrow(INVOCATION_EXCEPTION)
          .when(matcherSpy)
          .isSamhsaIcdProcedure(any(ProcedureComponent.class));
    return matcherSpy;
}
```

Reply:
I believe there's only ever 2 tests that set this up for one specific method at a time (if there would have been 3, I likely would have made it a parameterized test), so I don't think moving it to a private method would really help too much because there are many more that set it up for different methods that wouldn't be able to use that same private method.